nexus/src/app/background/tasks/inventory_collection.rs

```diff
@@ -132,19 +133,39 @@ async fn inventory_activate(
         })
         .collect::<Vec<_>>();
 
-    // Find clickhouse-admin-keeper clients
-    let keeper_admin_clients = resolver
-        .lookup_socket_v6(ServiceName::ClickhouseAdminKeeper)
+    // Find clickhouse-admin-keeper servers if there are any.
+    let keeper_admin_clients = match resolver
+        .lookup_all_socket_v6(ServiceName::ClickhouseAdminKeeper)
         .await
-        .context("looking up ClickhouseAdminKeeper addresses")
-        .into_iter()
-        .map(|sockaddr| {
-            let url = format!("http://{}", sockaddr);
-            let log =
-                opctx.log.new(o!("clickhouse_admin_keeper_url" => url.clone()));
-            clickhouse_admin_keeper_client::Client::new(&url, log)
-        })
-        .collect::<Vec<_>>();
+    {
+        Ok(sockaddrs) => sockaddrs
+            .into_iter()
+            .map(|sockaddr| {
+                let url = format!("http://{}", sockaddr);
+                let log = opctx
+                    .log
+                    .new(o!("clickhouse_admin_keeper_url" => url.clone()));
+                clickhouse_admin_keeper_client::Client::new(&url, log)
+            })
+            .collect::<Vec<_>>(),
+        Err(err) => match err {
```

Comment:
Piggy-backing on the discussion on an earlier version of this PR - do you think in the fullness of time (i.e., when all systems are running multinode clickhouse) we should stop swallowing errors here and always bubble them out? If so, is that worth an omicron issue + comment referring to it so we don't forget?

Reply:
Maybe? I'm honestly not sure. One issue is that we don't plan to enable clickhouse-cluster without running reconfigurator first. That means that after RSS there will still be a small gap where this will fail and prevent the rest of inventory from being collected. In any case, I do think it warrants at least a comment. Happy to open an issue as well.

Reply:
Done in [94904b1](https://github.com/oxidecomputer/omicron/pull/6950/commits/94904b1b5e11212c65ff3fd41d2f35e5df3759b8)